### DISCUSSION TITLE
How to get information of sender whose email was forwarded?

### DISCUSSION THREAD

**Discussion by ArunaAcharya** [mjs/imapclient#567] [General]
I love using Imap library and it is very helpful thank you so much for that, but I just came across one problem and I wanted to know if there is any solution for that. Here is the problem: 
let's say I have an email id **A**, **B** and n number of **C**. I am using Imap in email **A** to extract data. n number of C sends an email to B and B forwards all of that email to A. Is it possible to extract details of C when processing A emails?

**Comment by mjs** [comment]
This isn't really an IMAP question and it depends somewhat on how the forwarding is done, as there are different forwarding strategies.

If the emails are forwarded in a standard email client there will be headers and other changes to the email which might reveal with original sender. I advise you learn a little about the structure of emails and then take a look at some of the raw content of emails that have been forwarded.

The spec with details of how emails are structured is here: https://www.rfc-editor.org/rfc/rfc2822.html

The `email` package in the standard library can be used to parse and inspect emails. You can get the raw email content with IMAPClient by calling `fetch` with `RFC822` and then parse it with the functionality in the `email.parser` package.
